cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaClusterRebalanceAssemblyOperator.java

```diff
@@ -282,30 +289,41 @@ public void onClose(KubernetesClientException e) {
-                                            return Future.succeededFuture();
-                                        }
-                                    }).mapEmpty(), exception -> {
+                            .compose(freshClusterRebalance -> {
+                                if (freshClusterRebalance != null) {
+                                    return updateStatus(freshClusterRebalance, desiredStatus, clusterRebalanceOperator, null)
+                                            .compose(c -> {
+                                                log.info("{}: State updated to [{}] with annotation {}={} ",
+                                                        reconciliation,
+                                                        c.getStatus().getConditions().get(0).getType(),
+                                                        ANNO_STRIMZI_IO_REBALANCE,
+                                                        rebalanceAnnotation(c));
+                                                if (hasRebalanceAnnotation(c)) {
+                                                    log.debug("{}: Removing annotation {}={}", reconciliation, ANNO_STRIMZI_IO_REBALANCE,
+                                                            rebalanceAnnotation(c));
+                                                    KafkaClusterRebalance patchedClusterRebalance = new KafkaClusterRebalanceBuilder(c)
+                                                            .editMetadata().removeFromAnnotations(ANNO_STRIMZI_IO_REBALANCE).endMetadata().build();
+
+                                                    return clusterRebalanceOperator.patchAsync(patchedClusterRebalance);
+                                                } else {
+                                                    log.info("{}: No annotation {}", reconciliation, ANNO_STRIMZI_IO_REBALANCE);
+                                                    return Future.succeededFuture();
+                                                }
+                                            }).mapEmpty();
+
+                                } else {
+                                    return Future.succeededFuture();
+                                }
+                            }, exception -> {
                                     log.error("{}: Status updated to [NotReady] due to error", exception);
                                     return updateStatus(clusterRebalance, new KafkaClusterRebalanceStatus(), clusterRebalanceOperator, exception)
                                             .mapEmpty();
                                 });
-                });
+                }, exception -> {
+                        log.error("{}: Status updated to [NotReady] due to error", exception);
+                        return updateStatus(clusterRebalance, new KafkaClusterRebalanceStatus(), clusterRebalanceOperator, exception)
```

Comment:
What makes the Status `NotReady`? The Rebalance status is empty?

Reply:
The status utils class will fill the status with the information got from the `exception` as last parameter.